File: python/3진법뒤집기.py
def solution(n):
    answer = 0
    temp = ''
    if n < 3:
        return n
    
    while n != 0:
        temp = str(n%3) + temp
        n = int(n/3)
        if n < 3:
            temp = str(n) + temp
            break
    
    temp = temp[::-1]
    answer = int(temp[0])
    for i in range(len(temp)-1):
        answer = answer * 3 + int(temp[i+1])
    
    return answer

print(solution(1))
print(solution(2))
print(solution(3))
print(solution(4))
print(solution(5))
print(solution(6))

File: JavaScript/3진법뒤집기.js
function solution(n) {
    let str = '';
    while (n > 2) {
        str = n % 3 + str;
        n = Math.floor(n / 3);
    }
    let reverse = (n + str).split('').reverse().join('');
    let q = 0, answer = 0;
    for (let i = reverse.length - 1; i >= 0; i--) {
        answer += 3 ** q * reverse[i];
        q += 1;
    }
    return answer;
}
